code/ship/ship.cpp

```diff
@@ -9983,65 +9965,70 @@ float ship_get_thruster_status(object* obj, man_thruster* mtp) {
 	return val;
 }
 
-
-void ship_do_thruster_sounds(object* obj) {
-	if (Cmdline_freespace_no_sound)
-		return;
-
+// this function doesn't just play sounds but also keeps track of the time the thruster started firing
+void ship_do_thruster_sounds(object *obj)
+{
 	int num = obj->instance;
 	ship* shipp = &Ships[num];
 	ship_info* sip = &Ship_info[Ships[num].ship_info_index];
 
-	for (int i = 0; i < sip->num_maneuvering; i++) {
-		man_thruster* mtp = &sip->maneuvering[i];
+	// check whether we can play sounds in general and also on this ship
+	bool play_sound = !Cmdline_freespace_no_sound &&
+		(obj != Player_obj || Play_thruster_sounds_for_player);
 
-		float activated = ship_get_thruster_status(obj, mtp);
+	// make sure there are enough thruster instances
+	while (shipp->rcs_activity.size() < sip->rcs_thrusters.size())
+		shipp->rcs_activity.emplace_back(TIMESTAMP::invalid(), -1, 0.0f);
 
-		vec3d thruster_pos;
-		vm_vec_unrotate(&thruster_pos, &mtp->pos, &obj->orient);
-		vm_vec_add2(&thruster_pos, &obj->pos);
+	for (int i = 0; i < static_cast<int>(sip->rcs_thrusters.size()); i++)
```

Comment:
This should be a `size_t i` no? Wouldn't need the cast then and I think the other usages would still work as well.

Reply:
Yeah good point